### PR TITLE
fix(prover): Use SNARK wrapper VK for picking prover jobs

### PR DESCRIPTION
**Pull request by ly0va**
# What ❔

Prover and Witgen now use snark wrapper VKs to pick their jobs.

## Why ❔

`prover_protocol_versions` table had SNARK wrapper VKs, but prover and witgens picked jobs based on FRI scheduler VKs.

Since if one changes the other also does, we can use the SNARK one instead, since it's also used in `eth_sender`.

### CODE REVIEW

File: core/lib/zksync_core/src/lib.rs
Comment (StanislavBreadless) [inline core/lib/zksync_core/src/lib.rs:142]: Probably typo
```suggestion
        vk_hash, first_l1_verifier_config.recursion_scheduler_level_vk_hash,
```

File: prover/vk_setup_data_generator_server_fri/src/commitment_utils.rs
Comment (perekopskiy) [inline prover/vk_setup_data_generator_server_fri/src/commitment_utils.rs:28]: There is a JSON with snark wrapper vk (vk_setup_data_generator_server_fri/data/snark_verification_scheduler_key.json), I think you should calculate hash based on this key like it's done for other hashes and not just take it from env.

Comment (ly0va) [reply prover/vk_setup_data_generator_server_fri/src/commitment_utils.rs:28]: Unfortunately, the VK encodings that we use before hashing in Verifier.sol and in zkevm_test_harness are different (not only the order of fields is different, but also Verifier adds an extra field).

Unless we want to do a large copy-paste the hashes would be different. I think we should leave this for now and add actually computing the vk commitment later.

Comment (perekopskiy) [reply prover/vk_setup_data_generator_server_fri/src/commitment_utils.rs:28]: This is really unfortunate, I'm ok if we merge it to `boojum-integration` as is to unblock testing, but we should fix it before merging to main.

File: prover/vk_setup_data_generator_server_fri/src/commitment_utils.rs
Comment (mm-zk) [inline prover/vk_setup_data_generator_server_fri/src/commitment_utils.rs]: does this compile?

Comment (ly0va) [reply prover/vk_setup_data_generator_server_fri/src/commitment_utils.rs]: Already removed, was a merge artifact or something

File: prover/vk_setup_data_generator_server_fri/data/snark_verification_scheduler_key.json
Comment (mm-zk) [inline prover/vk_setup_data_generator_server_fri/data/snark_verification_scheduler_key.json:21]: please revert these changes.

File: core/lib/zksync_core/src/lib.rs
Comment (mm-zk) [inline core/lib/zksync_core/src/lib.rs:129]: I'm introducing a flag in https://github.com/matter-labs/zksync-era/pull/158 - so you could depend on it - this way we'd still be backwards compatible.

**Review by mm-zk — APPROVED** [review]

**Pull request merged**
